### ISSUE TITLE
Found unparsable section errors for postgres dialect

### ISSUE THREAD

**Issue by dnim** [sqlfluff/sqlfluff#2943]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When I'm running `sqlfluff lint .` against the next file I have the next errors:
```
DROP ROLE IF EXISTS supervisor;
DROP ROLE IF EXISTS nurse;
DROP ROLE IF EXISTS family_member;
CREATE ROLE supervisor;
CREATE ROLE nurse;
CREATE ROLE family_member;
```

```
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'DROP ROLE                                     
                       | IF EXISTS supervisor'
L:   4 | P:   1 |  PRS | Line 4, Position 1: Found unparsable section: 'DROP ROLE
                       | IF EXISTS nurse'
L:   5 | P:   1 |  PRS | Line 5, Position 1: Found unparsable section: 'DROP ROLE
                       | IF EXISTS family_member'
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'CREATE
                       | ROLE supervisor'
L:   7 | P:   1 |  PRS | Line 7, Position 1: Found unparsable section: 'CREATE
                       | ROLE nurse'
L:   8 | P:   1 |  PRS | Line 8, Position 1: Found unparsable section: 'CREATE
                       | ROLE family_member'
```

What can be the reason? 

### Expected Behaviour

No errors are found

### Observed Behaviour

Errors found
```
DROP ROLE IF EXISTS supervisor;
DROP ROLE IF EXISTS nurse;
DROP ROLE IF EXISTS family_member;
CREATE ROLE supervisor;
CREATE ROLE nurse;
CREATE ROLE family_member;
```

```
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'DROP ROLE                                     
                       | IF EXISTS supervisor'
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'CREATE
                       | ROLE supervisor'

```

### How to reproduce

run sqlfluff against the next sql:
```
DROP ROLE IF EXISTS supervisor;
DROP ROLE IF EXISTS nurse;
DROP ROLE IF EXISTS family_member;
CREATE ROLE supervisor;
CREATE ROLE nurse;
CREATE ROLE family_member;
```

```
L:   3 | P:   1 |  PRS | Line 3, Position 1: Found unparsable section: 'DROP ROLE                                     
                       | IF EXISTS supervisor'
L:   6 | P:   1 |  PRS | Line 6, Position 1: Found unparsable section: 'CREATE
                       | ROLE supervisor'

```

### Dialect

postgres

### Version

0.11.2

### Configuration

[sqlfluff]
dialect = postgres
exclude_rules = L057

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
For the `DROP ROLE` this is simply not defined in Postgres (nor in the ANSI dialect it inherits from).

Should be easy enough to add this. It IS defined in our Exasol dialect and think we should be able to pretty much copy and paste that to ANSI (perhaps without the `CASCADE` part) and Postgres will automatically inherit it to. Then just add a test case and then that's done.

For the `CREATE ROLE` looks to be a bug in Postgres definition.

At the moment it has this:

https://github.com/sqlfluff/sqlfluff/blob/a1385c1d759b9a7ef099d6b842de78b393bf149c/src/sqlfluff/dialects/dialect_postgres.py#L1095-L1104

This basically takes the ANSI definition and adds a bit at the end. However it doesn't really define that properly. The `Anything()` has to match something as not optional and doesn't with your statements. Also no need for extra `Sequence` in there.

Doing this instead, fixes it:

```py
    match_grammar = ansi.CreateRoleStatementSegment.match_grammar.copy(
        insert=[
            Ref.keyword("WITH", optional=True),
            # Very permissive for now. Anything can go here.
            Anything(optional=True),
        ],
    )
```

However, I think this trying to be too clever to be honest. The ANSI `CreateRoleStatementSegment` is fairly simple, so don't think we benefit from this `insert` way of doing it and my personal preference would be to just define the whole segment here (even though it's a small bit of repetition):

```py
    match_grammar = Sequence(
        "CREATE",
        "ROLE",
        Ref("ObjectReferenceSegment"),
        Ref.keyword("WITH", optional=True),
        # Very permissive for now. Anything can go here.
        Anything(optional=True),
    )
```

I think that's just clearer. And is more inline with what we do. So that's the change I'd do.

The `insert` method is more useful for more complex statements, or ones that are liable to change a lot, so you don't want to duplicate and have to maintain twice.

If you are willing to do a PR, then hopefully that'll set you on the track. We also just need a test case or two for these grammars added to the `test\fixtures\dialects\postgres` folder.

There's also a Wiki here that would be useful: https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes

**Comment by dnim** [comment]
Hi @tunetheweb! 
Thank you for so detailed intro and hints! 

I'll try to do PR on this week

**Comment by dnim** [comment]
@tunetheweb can you please provide a link on the trastable resource for SQL ANSI (2016 or 2019)? Tried to google but didn't get where is the original documentation

**Comment by tunetheweb** [comment]
Unfortunately you have to pay ANSI a lot of money for that. So we generally just implement what's common used in a few SQLs instead of strictly abiding by the ANSI standard.

**Comment by dnim** [comment]
oh :( that's pity 

I'll use cockroach db as an a basement. https://www.cockroachlabs.com/docs/stable/sql-grammar.html#select_stmt
